viewer-prototype/src/browser/trace-viewer/components/timegraph-output-component.tsx

```diff
@@ -19,20 +19,29 @@ import { TspDataProvider } from './data-providers/tsp-data-provider';
 import { ReactTimeGraphContainer } from "./utils/timegraph-container-component";
 import { OutputElementStyle } from 'tsp-typescript-client/lib/models/styles';
 
+
+
+
 type TimegraphOutputProps = AbstractOutputProps & {
     addWidgetResizeHandler: (handler: () => void) => void;
 }
 
 type TimegraohOutputState = AbstractOutputState & {
     timegraphTree: TimeGraphEntry[];
+    isClicked: boolean;
+    isTooltipOpened: boolean;
 }
 
 export class TimegraphOutputComponent extends AbstractTreeOutputComponent<TimegraphOutputProps, TimegraohOutputState> {
+    [x: string]: any;
     private totalHeight: number = 0;
     private rowController: TimeGraphRowController;
     private chartLayer: TimeGraphChart;
     private vscrollLayer: TimeGraphVerticalScrollbar;
     private horizontalContainer: React.RefObject<HTMLDivElement>;
+    private toolTipObject: { [key: string]: string } = {};
```

Comment:
Here I'm not sure, but should the toolTipObject be part of the state? Maybe a more react expert can tell, but fwiu, anything mutable that has effect on how the component will be rendered should be part of the state. And calling setState with the new tooltip set wil automatically re-render the component, so you don't have to call forceUpdate after setting this field, just setting the state does it. But I may be wrong...